neuro_san/internals/run_context/langchain/core/langchain_run_context.py

```diff
@@ -247,42 +247,33 @@ def create_agent_with_fallbacks(self, prompt_template: ChatPromptTemplate) -> Ag
         """
         Creates an agent.
         :param prompt_template: The ChatPromptTemplate to use for the agent
         :param llm: The BaseLanguageModel to use for the agent
         :return: An Agent (Runnable)
         """
         # Initialize our return value
-        agent: Agent = None
+        agent: Runnable = None
 
+        # Determine how complex the meat of our agent chain will be
+        meat: Runnable = llm
         if len(self.tools) > 0:
-            agent = create_tool_calling_agent(llm, self.tools, prompt_template)
-
-            # The above call creates a chain in this order:
-            #   first:  RunnablePassthrough
-            #   middle: prompt
-            #           llm_with_tools
-            #   last:   ToolsAgentOutputParser
-            #
-            # ... we need to mess with that a bit
-
-            # Replace the output parser from the call above.
-            # Per empirical experience, this is "last".
-            agent.last = JournalingToolsAgentOutputParser(self.journal)
-        else:
-            # This uses LangChain Expression Language (LCEL), which enables a functional, pipeline-style composition
-            # using "|". Here, we pass `agent_scratchpad` in the input message, but since we don't explicitly assign it
-            # to `intermediate_steps` (as done in `create_tool_calling_agent`), it remains unused by the prompt.
-            #
-            # In contrast, `create_tool_calling_agent` can be written in LCEL as
-            # RunnablePassthrough | prompt | llm_with_tools | ToolsAgentOutputParser
-            # where RunnablePassthrough `agent scratchpad` convert (AgentAction, tool output) tuples into ToolMessages.
-            #
-            # By skipping this step, our agent functions as a pure LLM-driven system with a defined role,
-            # without tool invocation logic influencing its decision-making.
-            agent = prompt_template | llm | JournalingToolsAgentOutputParser(self.journal)
+            meat = create_agent(model=llm, tools=self.tools)
+
+        # This uses LangChain Expression Language (LCEL), which enables a functional, pipeline-style composition
+        # using "|". Here, we pass `agent_scratchpad` in the input message, but since we don't explicitly assign it
+        # to `intermediate_steps` (as done in the old `create_tool_calling_agent`), it remains unused by the prompt.
+        #
+        # In contrast, the old `create_tool_calling_agent` can be written in LCEL as
+        # RunnablePassthrough | prompt | llm_with_tools | ToolsAgentOutputParser
+        # where RunnablePassthrough `agent scratchpad` convert (AgentAction, tool output) tuples into ToolMessages.
+        #
+        # By skipping this step, our agent functions as a pure LLM-driven system with a defined role,
+        # without tool invocation logic influencing its decision-making.
+
+        agent = RunnablePassthrough() | prompt_template | meat | JournalingToolsAgentOutputParser()
```

Comment:
Per the comment this was more or less what the single agent chain that we used ended boiling down to.

There had been more args passed to RunnablePassthrough, but those don't seem to be needed any more AFAICT.

Also, worth noting that for now we still need the end-stage JournalingToolsAgentOutputParser, but we no longer have to do any journaling from that guy. He just ends up being an output translator so this code can consume.   It's possible we can get rid of him entirely in a subsequent PR.